app/services/info_request_batch_zip.rb

```diff
@@ -106,8 +107,8 @@ def prepare_foi_attachment(attachment)
     path = [
       base_path(message.info_request),
       sent_at,
-      'attachments',
-      attachment.filename
+      "attachments-#{message.id}",
+      "#{idx}_#{attachment.filename}"
```

Comment:
I was hoping to avoid manipulating the attachment filenames. Its really confusing when they're different to what we show in the app.

How about doing something like:

```ruby
# …
begin
  attachment = prepare_foi_attachment(attachment)
rescue ZipTricks::PathSet::Conflict
  attachment = prepare_foi_attachment_with_part_namespace(attachment)
end
# …

def prepare_foi_attachment_with_namespace(attachment)
  # …
    path = [
      base_path(message.info_request),
      sent_at,
      "attachments-#{message.id}",
      attachment.url_part_number,
      attachment.filename
    ].join('/')
  # …
end
```

That way we mostly just have

```
attachments-123/
  foo.txt
  bar.txt
```

but in the odd case where there are duplicates, we get

```
attachments-123/
  1/
    foo.txt
  2/
    foo.txt
```

Reply:
I think with this suggestion, we would actually get 
```
attachments-123/
  foo.txt
  2/
    foo.txt
```
as the first attachment wouldn't raise an error.

Reply:
Ah, yeah, maybe the implementation needs a little bit of thinking, but the point still stands.

If it ends up being too hard, maybe just use `url_part_number` as a prefix rather than a random ID.

Reply:
Discovered an option in ZipTricks which handles automatic renaming of duplicate files:

With 54d809cabea7490b2f558d43bed5b85e74894bb0 we would get:
```
attachments-123/
  foo.txt
  foo (2).txt
```

@garethrees how is that?

Reply:
Oh nice! Yeah, that keeps things simple. Seems like a fairly uncommon case so lets just use this for now given its near zero effort.

Reply:
BTW can be tested locally with an existing email fixture file `./script/mailin < spec/fixtures/files/incoming-request-attach-attachments.email`